executor.py

```diff
@@ -74,8 +81,25 @@ def search(
             match_args.update(parameters)
 
         match_args = SimpleIndexer._filter_parameters(docs, match_args)
-
-        docs.match(self._storage, **match_args)
+        left_trav_path = match_args.get('traversal_ldarray')
+        # if it's 'r' we would just be duplicating
+        if left_trav_path and left_trav_path != 'r':
```

Comment:
if there's `traversal_ldarray` or `traversal_rdarray` args with value = 'r', we would still need to remove these args, so that deprecated warning is not thrown. Right now this case is not handled

Reply:
Yes, good catch. Done